### DISCUSSION TITLE
Support for Alternative Billing APIs

### DISCUSSION THREAD

**Discussion by opentechiz** [hyochan/react-native-iap#3041] [Polls]
### Discussed in https://github.com/dooboolab-community/react-native-iap/discussions/2678

<div type='discussions-op-text'>

<sup>Originally posted by **opentechiz** January 24, 2024</sup>
Dear,

Can you please help me about this question : Does the lib support the Alternative Billing APIs ? https://developer.android.com/google/play/billing/alternative/alternative-billing-with-user-choice-in-app

Best Regard,
Hoa</div>

**Comment by hyochan** [comment]
We are not supporting this feature. I believe there was an attempt in #2699, but it doesn't seem to be completed. The writer is @Ankit-R.

For more details, you can refer to the following link: [Google Play Billing Library Migration](https://developer.android.com/google/play/billing/migrate-gpblv7?_gl=1*1a8qge7*_up*MQ..*_ga*MzE3MTI3OTY1LjE3MTgxODAyMjE.*_ga_6HH9YJMN9M*MTcxODE4MDIyMC4xLjAuMTcxODE4MDIyMC4wLjAuMA..#handle-alternative-billing).

**Comment by Ritorna** [comment]
@hyochan as you're already working on the migration to the Billing Library V7, is there a possibility for you to also implement the User Choice Billing feature? I'm quite sure that this feature will be highly requested, once more and more developers are looking into adapting it to reduce the amount of fees collected by google.

**Comment by Syedh30** [comment]
@hyochan I have updated the user choice billing In this PR. [#2803 ](https://github.com/dooboolab-community/react-native-iap/pull/2803) works fine for me. Please check

**Comment by halukakin** [comment]
I think this is an important feature as ios and android will start supporting alternative payments in more regions in the coming future.
